lib/streamlit/runtime/memory_session_storage.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, MutableMapping, Optional
+
+from cachetools import TTLCache
+
+from streamlit.runtime.session_manager import SessionInfo, SessionStorage
+
+
+class MemorySessionStorage(SessionStorage):
+    """A SessionStorage that stores sessions in memory.
+
+    At most maxsize sessions are stored with a TTL of ttl seconds. This class is really
+    just a thin wrapper around cachetools.TTLCache that complies with the SessionStorage
+    protocol.
+    """
+
+    # NOTE: The defaults for maxsize and ttl are chosen arbitrarily for now. These
+    # numbers are reasonable as the main problems we're trying to solve at the moment are
+    # caused by transient disconnects that are usually just short network blips. In the
+    # future, we may want to increase both to support use cases such as saving state for
+    # much longer periods of time. For example, we may want session state to persist if
+    # a user closes their laptop lid and comes back to an app hours later.
+    def __init__(
+        self,
+        maxsize: int = 128,
+        ttl: int = 5 * 60,  # 5 minutes
```

Comment:
Can you add a docstring explaining what maxsize and ttl do? (Even if it's just a copy-paste from TTLCache.) In particular, what is the unit of `ttl`? (It looks like it's seconds - maybe change the name to `ttl_seconds`?)

Also, this _almost certainly doesn't matter_, but can `ttl` be a float? (Is `int` the proper type here?)

Reply:
Can do -- I'll add a docstring for this method and rename `ttl` -> `ttl_seconds`.

As for the type of `ttl`, usage in the cachetools documentation all happened to be using ints. Testing things out in the repl, it seems like floats are accepted but actually truncated (surprising to me, but 🤷), so I guess we might as well keep the type `int` given that behavior.